pyproject.toml

```diff
@@ -0,0 +1,180 @@
+# From the pip documentation:
+
+# Fallback Behaviour
+# If a project does not have a pyproject.toml file containing a build-system section,
+# it will be assumed to have the following backend settings:
+
+# [build-system]
+# requires = ["setuptools>=40.8.0", "wheel"]
+# build-backend = "setuptools.build_meta:__legacy__"
+
+# TODO: add appropriate build-system section
+[build-system]
+# TODO: specify minimum version of setuptools otherwise scikits.odes, NumPy, and others
+# will fail to install
+requires = [
+    "setuptools",
+    "wheel",
+    "casadi>=3.6.0; platform_system!='Windows'",
+    "cmake; platform_system=='Linux'",
+    ]
```

Comment:
Adding this works, but I am unsure whether this is the right or recommended approach. Adding `cmake` for GNU/Linux helped fix the issues with building editable wheels, allowing all tests to execute—and `casadi` was added as a build-time requirement here to work around the issue of isolated builds not working generally (CMake could not find casadi_DIR otherwise). This way, wheels can now be built both outside and within virtual environments without the need for `--no-build-isolation`. I am not very familiar with the configuration steps (will read up on them in the coming days) for CMakeLists.txt, however, if we can point the correct site-packages directory for `casadi` at the time of isolation, it should fix these issues. @arjxn-py's PR also had a similar issue, but the approach there was to use `setuptools.build_meta:__legacy__`, which is the default build-backend if this configuration value is unspecified.

Reply:
Resolving this as I think this was/is the right approach: in our case, we are linking against `casadi` at build time for our setup procedure with CMake regardless of whether build isolation is enabled or not. We should not be using the legacy build-backend